From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Wed, 19 Jan 2011 14:01:46 +0100
Subject: [PATCH] HACK: use $JARPREFIX instead of $JAVAUNIXJARDIR to look for unix.jar at runtime.

JAVAUNIXJARDIR cannot be used here because it is used at compile time and has
to start with $(SYSROOT).

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 Makefile |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 65bff86..8ebb3d3 100644
--- a/Makefile
+++ b/Makefile
@@ -88,7 +88,7 @@ translations/en_GB.po: $(SRCDIR)/*.java $(SRCDIR)/dbus/*.java $(SRCDIR)/dbus/exc
 	sed -n '/_(/s/.*_("\([^"]*\)").*/\1/p' $^ | sort -u | sed 's/\(.*\)/msgid "\1"\nmsgstr "\1"/' >> $@
 
 libdbus-java-$(VERSION).jar: .classes
-	echo "Class-Path: ${JAVAUNIXJARDIR}/unix.jar ${JAVAUNIXJARDIR}/hexdump.jar ${JAVAUNIXJARDIR}/debug-$(DEBUG).jar" > Manifest
+	echo "Class-Path: ${JARPREFIX}/unix.jar ${JARPREFIX}/hexdump.jar ${JARPREFIX}/debug-$(DEBUG).jar" > Manifest
 	(cd classes; $(JAR) -cfm ../$@ ../Manifest org/freedesktop/dbus/*.class org/freedesktop/*.class org/freedesktop/dbus/types/*.class org/freedesktop/dbus/exceptions/*.class *localized*class)
 dbus-java-test-$(VERSION).jar: .testclasses
 	echo "Class-Path: ${JARPREFIX}/libdbus-java-$(VERSION).jar" > Manifest
@@ -141,7 +141,7 @@ doc/api/index.html: $(SRCDIR)/*.java $(SRCDIR)/dbus/*.java .doc
 	$(DOCBOOKTOMAN) $< > $@
 	
 bin/%: %.sh .bin
-	sed 's,\%JARPATH\%,$(JARPREFIX),;s,\%JAVAUNIXJARPATH\%,$(JAVAUNIXJARDIR),;s,\%JAVAUNIXLIBPATH\%,$(JAVAUNIXLIBDIR),;s,\%VERSION\%,$(VERSION),;s,\%DEBUG\%,$(DEBUG),;s,\%JAVA\%,$(JAVA),' < $< > $@
+	sed 's,\%JARPATH\%,$(JARPREFIX),;s,\%JAVAUNIXJARPATH\%,$(JARPREFIX),;s,\%JAVAUNIXLIBPATH\%,$(JAVAUNIXLIBDIR),;s,\%VERSION\%,$(VERSION),;s,\%DEBUG\%,$(DEBUG),;s,\%JAVA\%,$(JAVA),' < $< > $@
 
 win/%.bat: %.bat .win
 	sed 's,\%WINJARPATH\%,$(JARPREFIX),;s,\%WINUNIXJARPATH\%,$(JAVAUNIXJARDIR),;s,\%VERSION\%,$(VERSION),;s,\%DEBUG\%,$(DEBUG),' < $< > $@
-- 
1.7.2.3

